### PR TITLE
New plotting overview

### PR DESCRIPTION
**Pull request by joferkington**
I realize this is a bit last-minute and a big change, but I really feel like we were missing a good overview of the various plotting methods.

I've added a new Part 2 (and renamed the other sections) to cover this: http://nbviewer.ipython.org/url/geology.beer/scipy2015/tutorial/AnatomyOfMatplotlib-Part2-Plotting_Methods_Overview.ipynb

I've tried to make a lot of nice summary images of the most commonly-used plotting methods. The "full" gallery can be very overwhelming, so it's useful to give people a condensed version. Also, these

80% of the new Part 2 is just quickly looking those images so that people are vaguely aware of what's out there. The code to generate them is also there to serve as an example.   

The new section only goes over `bar`, `fill_between` and `imshow` in more detail.  It's not anywhere near as long as it looks at first glance.

### CODE REVIEW

**Comment by WeatherGod** [comment]
Could you clear out the outputs of the cells? I make it a habit to clear all cell outputs prior to saving.

File: AnatomyOfMatplotlib-Part4-Limits_Legends_and_Layouts.ipynb [hunk 1]
```diff
@@ -340,7 +348,51 @@
...
      "metadata": {},
-     "outputs": []
+     "outputs": [],
+     "prompt_number": null
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "Another common "
```
Comment (WeatherGod) [inline AnatomyOfMatplotlib-Part4-Limits_Legends_and_Layouts.ipynb]: ????

**Comment by WeatherGod** [comment]
wow, the new example images are fantastic! I haven't yet looked at the new notebook as a whole, but this is very promising. It is quite likely that some of this is going to get folded into the official documentation, right?

The length is getting concerning. I get two break periods. I am going to have to practice the timing on the flight tomorrow and make various polishes then.

The narrative is also getting incohesive. While parts 1 and 2 are very well tied up, the content covered by parts 3, 4 & 5 become somewhat covered by those parts (colors and other properties, for example). The presentation doesn't seem to lead up to anything. The high point is pretty much part 2 now. I start to wonder if it makes sense to put part 2 at part 5.

File: README.md [hunk 2]
```diff
@@ -10,17 +10,19 @@ for conferences and other opportunities for training.
...
+[Part 2: Plotting Methods]: http://nbviewer.ipython.org/github/WeatherGod/AnatomyOfMatplotlib/blob/master/AnatomyOfMatplotlib-Part2-Plotting_Methods_Overview.ipynb
 [Part 3: How To Speak MPL]: http://nbviewer.ipython.org/github/WeatherGod/AnatomyOfMatplotlib/blob/master/AnatomyOfMatplotlib-Part3-HowToSpeakMPL.ipynb
-[Part 4: Artists]: http://nbviewer.ipython.org/github/WeatherGod/AnatomyOfMatplotlib/blob/master/AnatomyOfMatplotlib-Part4-Artists.ipynb
-[Part 5: mpl_toolkits]: http://nbviewer.ipython.org/github/WeatherGod/AnatomyOfMatplotlib/blob/master/AnatomyOfMatplotlib-Part5-mpl_toolkits.ipynb
+[Part 4: Limits, Legends, and Layouts]: http://nbviewer.ipython.org/github/WeatherGod/AnatomyOfMatplotlib/blob/master/AnatomyOfMatplotlib-Part2-Limits_Legends_and_Layouts.ipynb
+[Part 5: Artists]: http://nbviewer.ipython.org/github/WeatherGod/AnatomyOfMatplotlib/blob/master/AnatomyOfMatplotlib-Part4-Artists.ipynb
+[Part 6: mpl_toolkits]: http://nbviewer.ipython.org/github/WeatherGod/AnatomyOfMatplotlib/blob/master/AnatomyOfMatplotlib-Part5-mpl_toolkits.ipynb
```
Comment (WeatherGod) [inline README.md]: links are wrong (part numbers)

File: AnatomyOfMatplotlib-Part4-Limits_Legends_and_Layouts.ipynb (hunk 1 above)
Comment (joferkington) [inline AnatomyOfMatplotlib-Part4-Limits_Legends_and_Layouts.ipynb]: Ah, sorry that slipped in there. I accidentally applied a stash and forgot I'd applied it...

File: README.md (hunk 2 above)
Comment (joferkington) [inline README.md]: Fixed in cfe52c3. Thanks for catching that!

**Comment by joferkington** [comment]
Thanks!  I definitely would like to work them in as official examples at some point.

I agree length is becoming an issue, but I actually think the narrative is quite clear, at the moment.  
Basically:

  1. Overview of matplotlib, figures, and axes (i.e. what you need to do to begin plotting)
  2. Overview of plotting methods
  3. Common controls for plotting methods (markers, colors, etc)
  4. Changing ticks, spines, etc (i.e. what goes on around the borders of the axes)
  5. Lower-level primitives
  6. Mpl toolkits

There are a few things that could be in other places (for example legends could be just about anywhere, and the colormap portion of Part 3 could conceivably be folded into Part 2).  However, I think things flow fairly well as they are.

On the length side, it's getting long, but I think it's still manageable.  

A lot of the time spent in the new Part 2 will be on the two exercises (probably 5-10 mins each).  The rest will probably take 15-20 minutes.  I don't intend to go over the example code for the images -- just have them there as a visual reference and quick mention.  Part 4 could be trimmed down a bit, and Part 5 could conceivably be skipped (though it would definitely be better to cover it).

On a side note, all of the outputs are cleared, as far as I can tell.  Am I missing one somewhere?

**Pull request merged**